### ISSUE TITLE
JSON data seems wrong for "post-details"

### ISSUE THREAD

**Issue by Smart-Ace-Designs** [Code-Pop/Real-World-Nuxt-3#4]
The JSON data from the repo looks like this for "post-details":

```json
{
      "id": "nuxt-page",
      "slug": "nuxt-page",
      "categoryId": "components",
      "title": "Rendering a Page",
      "content": "You can render a Nuxt page by using the `<NuxtPage>` component.\n\nYou have to put it in the `app.vue` file:\n```html\n<template>\n  <div>\n    <NuxtPage />\n  </div>\n</template>\n```\n\nIt will be able to inspect the current path and render the corresponding page in your **pages** folder."
},
```

However, you code is referencing "category" not "categoryId" (unless I missed some step in the video):
```vue
<template>
  <main>
    <template v-if="post">
      <h1>
        {{ post.title }}
        <CategoryLink :category="post.category" />
      </h1>
      <RenderMarkdown :source="post.content" />
    </template>
  </main>
</template>
```

The only way I could get it to work was to modify your JSON file as such:
```json
{
      "id": "nuxt-page",
      "slug": "nuxt-page",
      "categoryId": "components",
      "category": {
        "id": "components",
        "name": "Components",
        "slug": "components",
        "count": 5
      },
      "title": "Rendering a Page",
      "content": "You can render a Nuxt page by using the `<NuxtPage>` component.\n\nYou have to put it in the `app.vue` file:\n```html\n<template>\n  <div>\n    <NuxtPage />\n  </div>\n</template>\n```\n\nIt will be able to inspect the current path and render the corresponding page in your **pages** folder."
},
```

So maybe I missed something while following along, but it appears your JSON file is incorrect in terms of the video on VueMastery.

**Comment by tutorialcode** [comment]
That's how json-server works, the category info will show up in the product details result without any modification in db.json. If you are not getting that, please let us know the error you're getting if any, or elaborate more on how it's not working.

You can also take a look at the lesson's branch to make sure you have the same code for that page component.

**Comment by Smart-Ace-Designs** [comment]
Ah, I see the problem.  I took liberties with updating the package dependencies and updated "json-server" to version "1.0.0-alpha.23" per the the Nuxtr extension suggestion.  Once I switched back to version "0.17.4" it worked with the "db.json" you provided in the repo.

So the moral of the story is:
1. There appears to a breaking change with the latest json-server :frowning:
2. Don't mess with the dependency version suggested by the author when following a tutorial :flushed:

Sorry for false alarm.  Thank you for the tutorial and free weekend on VueMastery.
